app/src/main/java/ch/epfl/sdp/artificial_intelligence/Movement.java

```diff
@@ -1,10 +1,65 @@
-package ch.epfl.sdp.artificial_intelligence;
-
-public enum Movement {
-    LINEAR,
-    SINUSOIDAL,
-    CIRCULAR/*,
-    CURVED,
-    SMOOTH,
-    RANDOM*/
-}
+package ch.epfl.sdp.artificial_intelligence;
+
+public class Movement {
+    private float velocity;
+    private float acceleration;
+    private double orientation;
+    private MovementType movementType;
+    private double sinusAmplitude = 1;
+    private double sinusAngle;
+
+    public Movement() {
+        acceleration = 0;
+        velocity = 0;
+        orientation = 0;
+        movementType = MovementType.LINEAR;

```

Comment:
Shall we also initialize sinusAngle?

Reply:
We should yes, in this branch I only moved this code to another class, so I did not change anything. Normally it should already be initialized to 0, but we can do it explicitely.